### PR TITLE
2715_GameChannels_Complexity

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

File: client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts [hunk 1]
```diff
@@ -22,107 +32,121 @@ const initialState = {
...
-      newChannel.name = action.payload.name
-      newChannel.uniqid = action.payload.uniqid
-      newState.data[tab].channels.push(newChannel)
-      break
-
-    case ActionConstant.SET_SELECTED_CHANNEL:
-      newState.data[tab].selectedChannel = action.payload
-      break
-
-    case ActionConstant.DELETE_SELECTED_CHANNEL:
```
Comment (IanMayo) [inline client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts:120]: @lilitkarapetyan - if the `handler` is empty in the next next line, should we throw a warning/error?

File: client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts [hunk 2]
```diff
@@ -22,107 +32,121 @@ const initialState = {
...
+    // case ActionConstant.SET_SELECTED_CHANNEL:
+    //   newState.data[tab].selectedChannel = action.payload
+    //   break
+
+    // case ActionConstant.DELETE_SELECTED_CHANNEL:
+
+    //   channelIndex = newState.data[tab].channels.findIndex((channel: ChannelTypes) => channel.name === action.payload.name)
+    //   newState.data[tab].channels.splice(channelIndex, 1)
+    //   newState.data[tab].selectedChannel = ''
+    //   break
```
Comment (IanMayo) [inline client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts]: should the next block be in the `handlers` @lilitkarapetyan ?

File: client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts [hunk 3]
```diff
@@ -22,107 +32,121 @@ const initialState = {
...
-
-    case ActionConstant.DELETE_SELECTED_CHANNEL:
+  if (handler) {
+    handler(newState, action, tab)
+  }
 
-      channelIndex = newState.data[tab].channels.findIndex((channel: ChannelTypes) => channel.name === action.payload.name)
-      newState.data[tab].channels.splice(channelIndex, 1)
-      newState.data[tab].selectedChannel = ''
-      break
```
Comment (IanMayo) [inline client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts:125]: we don't need the following `switch`, do we?

File: client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts (hunk 1 above)
Comment (lilitkarapetyan) [reply client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts:120]: I will do add warrning.

File: client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts (hunk 2 above)
Comment (lilitkarapetyan) [reply client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts]: yes

File: client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts (hunk 3 above)
Comment (lilitkarapetyan) [reply client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts:125]: I plan to eliminate the swich statement in the final version.

**Review by IanMayo — APPROVED** [review]

**Pull request merged**